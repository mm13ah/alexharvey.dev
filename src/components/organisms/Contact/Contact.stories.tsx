import React from 'react';
import { StoryFn, Meta } from '@storybook/react';

import { Contact } from './Contact';

export default {
  title: 'Organisms/Contact',
  component: Contact,
  args: {},
} as Meta<typeof Contact>;

const Template: StoryFn<typeof Contact> = () => <Contact />;

export const Primary = Template.bind({});
Primary.storyName = 'Contact';
