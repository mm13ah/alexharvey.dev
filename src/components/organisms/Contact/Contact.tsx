import { useTheme } from 'next-themes';
import React, { useEffect, useState } from 'react';
import { BsEnvelope } from 'react-icons/bs';
import { SiGithub } from 'react-icons/si';
import { Background } from '../../atoms';

export const Contact = () => {
  const { theme } = useTheme();
  const [mounted, setMounted] = useState(false);
  useEffect(() => {
    setMounted(true);
  }, []);
  return (
    <Background
      id="contact"
      className={`bg-white dark:bg-gradient-to-r dark:from-gray-900 dark:via-stone-700
      dark:to-gray-900 p-10 items-center justify-evenly flex-col`}
    >
      <h2
        className={`text-neutral-900 text-center font-medium tracking-wider text-5xl mb-2
        w-full show-on-scroll dark:text-white`}
      >
        Get In Touch
      </h2>
      <BsEnvelope
        size={80}
        color={
          !mounted ? 'transparent' : theme === 'dark' ? 'white' : '#171717'
        }
        className="show-on-scroll"
      />
      <div className="flex flex-col h-full w-full items-center justify-center show-on-scroll">
        <h3 className="text-3xl">Alex Harvey</h3>
        <h4 className="font-light mt-2">aharvey7nr7@gmail.com</h4>
        <h4 className="font-light mt-1">alexharvey.dev</h4>
        <div className="flex gap-x-10 mt-3">
          <a href="https://github.com/mm13ah" target="_blank" rel="noreferrer">
            <SiGithub
              size={30}
              color={
                !mounted
                  ? 'transparent'
                  : theme === 'dark'
                  ? 'white'
                  : '#171717'
              }
            />
          </a>
        </div>
      </div>
    </Background>
  );
};
