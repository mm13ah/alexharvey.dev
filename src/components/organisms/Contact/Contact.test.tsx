import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Contact } from './Contact';

describe('Contact', () => {
  it('should render', () => {
    render(<Contact />);
  });
  it('should match a snapshot', () => {
    const tree = renderer.create(<Contact />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
