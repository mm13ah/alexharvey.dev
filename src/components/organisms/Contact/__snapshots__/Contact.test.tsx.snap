// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact should match a snapshot 1`] = `
<section
  className="w-screen min-h-screen flex bg-white dark:bg-gradient-to-r dark:from-gray-900 dark:via-stone-700
      dark:to-gray-900 p-10 items-center justify-evenly flex-col"
  id="contact"
>
  <h2
    className="text-neutral-900 text-center font-medium tracking-wider text-5xl mb-2
        w-full show-on-scroll dark:text-white"
  >
    Get In Touch
  </h2>
  <svg
    className="show-on-scroll"
    color="transparent"
    fill="currentColor"
    height={80}
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": "transparent",
      }
    }
    viewBox="0 0 16 16"
    width={80}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 4a2 2 0 0 1 2-2h12a2 2 0 0 1 2 2v8a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2V4zm2-1a1 1 0 0 0-1 1v.217l7 4.2 7-4.2V4a1 1 0 0 0-1-1H2zm13 2.383-4.758 2.855L15 11.114v-5.73zm-.034 6.878L9.271 8.82 8 9.583 6.728 8.82l-5.694 3.44A1 1 0 0 0 2 13h12a1 1 0 0 0 .966-.739zM1 11.114l4.758-2.876L1 5.383v5.73z"
    />
  </svg>
  <div
    className="flex flex-col h-full w-full items-center justify-center show-on-scroll"
  >
    <h3
      className="text-3xl"
    >
      Alex Harvey
    </h3>
    <h4
      className="font-light mt-2"
    >
      aharvey7nr7@gmail.com
    </h4>
    <h4
      className="font-light mt-1"
    >
      alexharvey.dev
    </h4>
    <div
      className="flex gap-x-10 mt-3"
    >
      <a
        href="https://github.com/mm13ah"
        rel="noreferrer"
        target="_blank"
      >
        <svg
          color="transparent"
          fill="currentColor"
          height={30}
          role="img"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "transparent",
            }
          }
          viewBox="0 0 24 24"
          width={30}
          xmlns="http://www.w3.org/2000/svg"
        >
          <title />
          <path
            d="M12 .297c-6.63 0-12 5.373-12 12 0 5.303 3.438 9.8 8.205 11.385.6.113.82-.258.82-.577 0-.285-.01-1.04-.015-2.04-3.338.724-4.042-1.61-4.042-1.61C4.422 18.07 3.633 17.7 3.633 17.7c-1.087-.744.084-.729.084-.729 1.205.084 1.838 1.236 1.838 1.236 1.07 1.835 2.809 1.305 3.495.998.108-.776.417-1.305.76-1.605-2.665-.3-5.466-1.332-5.466-5.93 0-1.31.465-2.38 1.235-3.22-.135-.303-.54-1.523.105-3.176 0 0 1.005-.322 3.3 1.23.96-.267 1.98-.399 3-.405 1.02.006 2.04.138 3 .405 2.28-1.552 3.285-1.23 3.285-1.23.645 1.653.24 2.873.12 3.176.765.84 1.23 1.91 1.23 3.22 0 4.61-2.805 5.625-5.475 5.92.42.36.81 1.096.81 2.22 0 1.606-.015 2.896-.015 3.286 0 .315.21.69.825.57C20.565 22.092 24 17.592 24 12.297c0-6.627-5.373-12-12-12"
          />
        </svg>
      </a>
    </div>
  </div>
</section>
`;
